hashicorp/consul
# NET-6821 Disable Terminating Gateway Auto Host Header Rewrite

### Description
- Exposing configuration to allow users of the terminating gateway to turn off host header rewrites at the service level 

### Testing & Reproduction steps
- build local consul 
- apply the following terminating gateway 
```
Kind = "terminating-gateway"
Name = "test-gateway"
Services = [
 {
   Name = "default-behavior"
 },
 {
   Name = "default-behavior"
   DisableAutoHostRewrite = true
 }
]

```

Observe the generated XDS and note that the default behavior is to auto host rewrite, but not when DisableAutoHostRewrite is set to true.

### Links



### PR Checklist

* [ X ] updated test coverage
